spec/src/main/asciidoc/chapters/language/expressions.adoc

```diff
@@ -466,40 +466,33 @@ A quantified comparison expression is satisfied if and only if:
+For any numeric type, or for any date or datetime type, there is a _natural order_ completely determined by the semantics of the type. Jakarta Query implementations must sort these types according to their natural order:
 
-Boolean values must be ordered so that `false < true` is satisfied.
+- smaller numeric values come before larger numeric values, and
+- earlier dates or datetimes come before later dates or datetimes.
 
-For other types, there is at least some freedom in the choice of order. Usually, the order is determined by the native semantics of the database. Note that:
+NOTE: For an implementation of Jakarta Query which targets clients written in Java, this natural order agrees with the order defined by the corresponding Java type from `java.lang`, `java.math`, or `java.time`.
 
-- Textual data is represented in Java as the type `java.lang.String`. Strings are in general ordered lexicographically, but the ordering also depends on the character set and collation used by the database server. Applications must not assume that the order agrees with the `compareTo()` method of `java.lang.String`. In evaluating an inequality involving string operands, an implementation of Jakarta Query is not required to emulate Java collation.
-
-- Binary data is represented in Java as the type `byte[]`. Binary data is in general ordered lexicographically with respect to the constituent bytes. However, since this ordering is rarely meaningful, this specification does not require implementations of Jakarta Query to respect it.
-
-- This specification does not define an order for the sorting of Java `enum` values, which is provider-dependent. An implementation of Jakarta Query might allow control over the order of `enum` values. For example, Jakarta Persistence allows this via the `@Enumerated` annotation.
+Furthermore, Jakarta Query implementations must sort boolean values so that `false < true` is satisfied.
 
-- This specification does not define an order for UUID values, which is provider-dependent.
+For other types, there is at least some freedom in the choice of order. Usually, the order is determined by the native semantics of the database.
 
-[WARNING]
+[CAUTION]
 ====
-When using NoSQL databases, sorting support varies by database type:
+For clients written in Java, note that:
 
-Key-value databases:: Sorting of results is not supported.
+- Textual data is represented in Java as the type `java.lang.String`. Strings are in general ordered lexicographically, but the ordering also depends on the character set and collation used by the database server. Applications must not assume that the order agrees with the `compareTo()` method of `java.lang.String`. In evaluating an inequality involving string operands, an implementation of Jakarta Query is not required to emulate Java collation.
```

Comment:
This one is solid. We need to cover the other cases.

Reply:
I would like you to reread the text, taking the perspective that this section is defining the semantics of _Jakarta Query_, and it's not saying anything about the mapping to _Java_ except in the two (non-prescriptive) admonitions. In fact, the second admonition points out places where the natural order in Jakarta Query is definitely _inconsistent_ with Java's `compareTo()`.